test/srp6a.test.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable no-fallthrough */
+import { BigInteger } from "jsbn";
+import { SRPConfig } from "../config";
+import { SRPParameters } from "../parameters";
+import { SRPRoutines } from "../routines";
+import { SRPClientSession } from "../session-client";
+import { SRPServerSession } from "../session-server";
+import {
+  bigIntegerToWordArray,
+  createVerifierAndSalt,
+  generateRandomString,
+  stringToWordArray,
+  wordArrayToBigInteger,
+} from "../utils";
+import { test } from "./tests";
+
+const testParameters = new SRPParameters();
+class SRP6aRoutines extends SRPRoutines {
+  public computeX(_I: string, s: BigInteger, P: string): BigInteger {
+    return wordArrayToBigInteger(
+      this.hash(
+        bigIntegerToWordArray(s),
+        this.hash(stringToWordArray(`${_I}:${P}`)),
+      ),
+    );
+  }
+}
+const TestConfig = new SRPConfig(testParameters, (p) => new SRP6aRoutines(p));
+
+test("#SRP6aSession success", (t) => {
```

Comment:
what's the difference between this and https://github.com/midonet/tssrp6a/blob/210ee15f41f00f503e988a30ee9e1be9c584fb1f/test/session.test.ts ?

Reply:
Not too much, I actually copied it from there :)
The main use case for this test is documentation reference.

Reply:
sounds like confusing to the user

Reply:
Yes, you are right. I had this doubt while working on this patch: why we just don't make default configuration strictly follow RFC and avoid this note? But our current default config is compatible with default config of nimbusds java implementation of SRP6a, so it also makes some sense.
    
I might be wrong, but I think that in our previous project while looking for JS implementation of SRP6a protocol we discarded one project because it did not work out of the box with nimbusds java server.

I think that the documentation should be very clear about this. Currently we just mention it in a vague way.
   
It also does not heart to have more usage examples.

Reply:
So this should be organized and made clear in some way, or it will be very confusing or not work for someone who tries to pick this lib up.

Is `SRP6aSession success` test the same? If so, only keep one.
If the difference is that one suite tests for SRP6a defaults, and the other for java-nimbus, than that should probably be made clear. E.g. we should rename both tests (filename too if appropiate), and add comments or references to the docu.

Reply:
I've added one fixup commit to the PR that tries to make everything even more clear.
The session.test.ts has many tests, so its name is fine I think.
We do mention both tests in the README.md and highlight the difference.
Please, notice, that srp6a.test.ts is slightly different test, since it uses different computeX() routine.
Please, read the doc and make sure that it is very clear :)

Reply:
thank you Dmitry, now I see what you are after.